### PR TITLE
Update NEST Homebrew installation instructions to changes in Homebrew…

### PR DESCRIPTION
**Pull request by heplesser**
Update NEST Homebrew installation instructions to changes in Homebrew repo structure.

Also add info on how to install from moste recent sources.

@tammoippen Could you check that I got it right?
@steffengraber Can you update the website once both @tammoippen and you are happy?

### CODE REVIEW

**Comment by tammoippen** [comment]
Same as #879: I am working to get nest into homebrew core. With #872 merged, there should be no barrier.

**Comment by heplesser** [comment]
@tammoippen The instructions we have on our website do not work, so brewsci is the only option. Since the integration into homebrew/core is beyond our control, I think we need to provide instructions that work for now. We certainly cannot have instructions on the website that do not work.

File: extras/userdoc/md/documentation/installation.md [hunk 1]
```diff
@@ -151,6 +151,11 @@ The easiest way to install NEST on a Mac is to install it via the Homebrew packa
 Options have to be appended, e.g. to install NEST with PyNEST run
 
     brew install nest --with-python
+    
+This will install the most recent release version of NEST. To build
+NEST from the most recent sources on Github, use
+
+    brew install nest --HEAD
```
Comment (tammoippen) [inline extras/userdoc/md/documentation/installation.md]: See https://github.com/brewsci/homebrew-science/issues/73 - does not work. Also with nest on homebrew-core the option `--HEAD` will not be provided, see https://github.com/Homebrew/homebrew-core/pull/21561#pullrequestreview-82487750 (open `Show outdated`).

File: extras/userdoc/md/documentation/installation.md [hunk 2]
```diff
@@ -137,7 +137,7 @@ The easiest way to install NEST on a Mac is to install it via the Homebrew packa
 
     1.  Add the homebrew/science tap by running 
     
-        `brew tap homebrew/science`
+        `brew tap brewsci/science`
```
Comment (tammoippen) [inline extras/userdoc/md/documentation/installation.md]: Hopefully temporary: need to be changed, when https://github.com/Homebrew/homebrew-core/pull/24908 is merged.

**Pull request merged**